File: atcoder/tyama_atcodercodefestival2017finalC.py
#!/usr/bin/python
n=input()
a=sorted(map(int,raw_input().split()))
b=[0]+[a[j]*(-1)**j%24 for j in range(n)]
s=24
for j in range(n):
 for k in range(j+1,n+1):s=min(s,(b[j]-b[k])%24,(b[k]-b[j])%24)
print s

File: atcoder/codefestival/tyama_atcodercodefestival2017finalC.rb
#!/usr/bin/ruby
n,*a=`dd`.split.map &:to_i;a.sort!
b=[0]+n.times.map{|j|a[j]*(-1)**j%24}
s=24
n.times{|j|(j+1..n).each{|k|s=[s,(b[j]-b[k])%24,(b[k]-b[j])%24].min}}
p s
